src/main/scala/is/hail/io/plink/PlinkLoader.scala

```diff
@@ -200,8 +137,59 @@ object PlinkLoader {
            |  Duplicate IDs: @1""".stripMargin, duplicateIds)
     }
 
-    val vds = parseBed(hc, bedPath, ids, annotations, signature, variants, nPartitions, a2Reference)
-    vds
-  }
+    val results = bedPaths.zip(bimPaths).map { case (bedPath, bimPath) =>
+      val variants = parseBim(bimPath, hc.hadoopConf, a2Reference)
+      val nVariants = variants.length
+      if (nVariants <= 0)
+        fatal(s".bim file does not contain any variants [$bimPath]")
 
+      info(s"Found $nVariants variants in BIM file [$bimPath].")
+
+      hc.hadoopConf.readFile(bedPath) { dis =>
+        val b1 = dis.read()
+        val b2 = dis.read()
+        val b3 = dis.read()
+
+        if (b1 != 108 || b2 != 27)
+          fatal(s"First two bytes of bed file do not match PLINK magic numbers 108 & 27 [$bedPath]")
+
+        if (b3 == 0)
+          fatal(s"BED file is in individual major mode. First use plink with --make-bed to convert file to SNP major mode before using Hail [$bedPath]")
+      }
+
+      val bedSize = hc.hadoopConf.getFileSize(bedPath)
+      if (bedSize != expectedBedSize(nSamples, nVariants))
+        fatal(s"BED file size does not match expected number of bytes based on BIM and FAM files [$bedPath]")
+
+      if (bedSize < nPartitions.getOrElse(hc.sc.defaultMinPartitions))
+        fatal(s"The number of partitions requested (${ nPartitions.getOrElse(hc.sc.defaultMinPartitions) }) is greater than the file size ($bedSize). [$bedPath]")
+
+      val variantsBc = sc.broadcast(variants)
+      sc.hadoopConfiguration.setInt("nSamples", nSamples)
+      sc.hadoopConfiguration.setBoolean("a2Reference", a2Reference)
+
+      sc.hadoopFile(bedPath, classOf[PlinkInputFormat], classOf[LongWritable], classOf[PlinkRecord],
+        nPartitions.getOrElse(sc.defaultMinPartitions))
```

Comment:
I think this is incorrect behavior, even though it's consistent with the other import methods. It could go in, but we should at least be cognizant.

Reply:
(applying npartitions to each part rather than the complete result)